egs/tedlium3/asr1/run.sh

```diff
@@ -0,0 +1,317 @@
+#!/bin/bash
+
+# Copyright 2019 Nagoya University (Masao Someki)
+#  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
+
+. ./path.sh
+. ./cmd.sh
+
+# general configuration
+backend=pytorch
+stage=-1       # start from -1 if you need to start from data download
+stop_stage=100
+ngpu=0         # number of gpus ("0" uses cpu, otherwise use gpu)
+debugmode=1
+dumpdir=dump   # directory to dump full features
+N=0            # number of minibatches to be used (mainly for debugging). "0" uses all minibatches.
+verbose=0      # verbose option
+resume=        # Resume the training from snapshot
+
+# feature configuration
+do_delta=false
+
+# network architecture
+# encoder related
+etype=vggblstmp     # encoder architecture type
+elayers=4
+eunits=1024
+eprojs=1024
+subsample=1_2_2_1_1 # skip every n frame from input to nth layers
+# decoder related
+dlayers=1
+dunits=1024
+# attention related
+atype=location
+adim=1024
+aconv_chans=10
+aconv_filts=100
+
+# hybrid CTC/attention
+mtlalpha=0.5
+
+# minibatch related
+batchsize=35
+maxlen_in=600  # if input length  > maxlen_in, batchsize is automatically reduced
+maxlen_out=150 # if output length > maxlen_out, batchsize is automatically reduced
+
+# optimization related
+sortagrad=0 # Feed samples from shortest to longest ; -1: enabled for all epochs, 0: disabled, other: enabled for 'other' epochs
+opt=adadelta
+epochs=10
+patience=3
+
+# rnnlm related
+lm_layers=2
+lm_units=650
+lm_opt=sgd        # or adam
+lm_sortagrad=0 # Feed samples from shortest to longest ; -1: enabled for all epochs, 0: disabled, other: enabled for 'other' epochs
+lm_batchsize=512  # batch size in LM training
+lm_epochs=20      # if the data size is large, we can reduce this
+lm_patience=3
+lm_maxlen=150     # if sentence length > lm_maxlen, lm_batchsize is automatically reduced
+lm_resume=        # specify a snapshot file to resume LM training
+lmtag=            # tag for managing LMs
+
+# decoding parameter
+lm_weight=1.0
+beam_size=20
+penalty=0.0
+maxlenratio=0.0
+minlenratio=0.0
+ctc_weight=0.3
+recog_model=model.acc.best # set a model to be used for decoding: 'model.acc.best' or 'model.loss.best'
+
+# scheduled sampling option
+samp_prob=0.0
+
+# bpemode (unigram or bpe)
+nbpe=500
+bpemode=unigram
+
+# exp tag
+tag="" # tag for managing experiments.
+
+. utils/parse_options.sh || exit 1;
+
+. ./path.sh
+. ./cmd.sh
+
+# Set bash to 'debug' mode, it will exit on :
+# -e 'error', -u 'undefined variable', -o ... 'error in pipeline', -x 'print commands',
+set -e
+set -u
+set -o pipefail
+
+train_set=train_trim_sp
+train_dev=dev_trim
+recog_set="dev test"
```

Comment:
Is it possible to keep both adaptation and legacy data directories, and just switch either option by setting the `train_set`, `train_dev`, and `recog_set` with corresponding comments? 
We could create both data directories in `local/prepare_data.sh`.
We can keep the current name for the legacy one, and name the adaptation directories as `train_adapt_trim_sp`, `dev_adapt_trim`, `dev_adapt`, and `test_adapt`

Reply:
For example,

```
# legacy setup
train_set=train_trim_sp
train_dev=dev_trim
recog_set="dev test"
# adaptation setup
#train_set=train_adapt_trim_sp
#train_dev=dev_adapt_trim
#recog_set="dev_adapt test_adapt"
```
and generate the both legacy and adaptation direcotries in `local/prepare_data.sh`.

Reply:
Okay.